### ISSUE TITLE
Logging each transfer info into separated file no longer supported?

### ISSUE THREAD

**Issue by enricovianello** [gridcf/gct#110]
We were used to keep separated all the gridftp server's transfer info with -log-transfer option.
But it seems that latest releases have removed this feature because the transfer log file remains empty.
Can you confirm this or -log-transfer option should it still work?

**Comment by enricovianello** [comment]
After reading code and commits, it may had stopped working after the removal of usage statistics collection support: https://github.com/gridcf/gct/commit/9a8f75e26261580638d7e36c0aa845b2617e4891#diff-7b8084ebaf25da156f409243079c6b76
could you confirm this?

**Comment by fscheiner** [comment]
@enricovianello: Thanks for the pointer, I didn't notice that so far.

But I think the relevant code was removed specifically here https://github.com/gridcf/gct/commit/9a8f75e26261580638d7e36c0aa845b2617e4891#diff-0d41fe5a384e8076fc8f2dc3424bc45a.

I wonder if `-log-transfer` can work without the Globus Usage Statistics and that it was accidently removed, because the information bits gathered and logged are very similar to identical for both from a quick check. If not, I'm not sure if that functionality will return, as it was/is more important to have that _phone home_ "feature" removed.

CCing @ellert, @matyasselmeci, @msalle here for their assessment.

BTW, `globus_i_gfs_log_transfer()` - the function used in a part of the above mentioned diff - is still in the source according to a [quick search](https://github.com/gridcf/gct/search?q=globus_i_gfs_log_transfer&unscoped_q=globus_i_gfs_log_transfer), so maybe this can work w/o the Globus Usage Statistics and we can just partially revert the above mentioned diff to reactivate that `-log-transfer` functionality.

**Comment by msalle** [comment]
I'm not sure, we should certainly have feedback from @ellert, but looking at the commit in [L10420-10424](https://github.com/gridcf/gct/commit/9a8f75e26261580638d7e36c0aa845b2617e4891#diff-0d41fe5a384e8076fc8f2dc3424bc45aL10420-L10424) and later [L10473](https://github.com/gridcf/gct/commit/9a8f75e26261580638d7e36c0aa845b2617e4891#diff-0d41fe5a384e8076fc8f2dc3424bc45aL10473) versus [L10492-10493](https://github.com/gridcf/gct/commit/9a8f75e26261580638d7e36c0aa845b2617e4891#diff-0d41fe5a384e8076fc8f2dc3424bc45aL10492-L10493) it looks like we might be able to roll back just the `-log-transfer` functionality.

**Comment by fscheiner** [comment]
Maybe in the meantime @enricovianello could locally test a patch that only restores the `-log-transfer` functionality? Unfortunately ATM I'm fully occupied with other tasks (GSI-OpenSSH w/HPN for SUSE), so cannot take care of such a patch for some time. So @enricovianello, do you think you could come up with a patch that would accomplish that? Then if @ellert is OK with it later, we could just include that patch in the next GCT (maintenance) version.

**Comment by fscheiner** [comment]
@enricovianello:
@ellert has partially reverted https://github.com/gridcf/gct/commit/9a8f75e26261580638d7e36c0aa845b2617e4891#diff-0d41fe5a384e8076fc8f2dc3424bc45a to restore the `log-transfer` functionality. If you're building your GridFTP server directly from the sources, could you please give it a try with these changes included and report back the result?

**Comment by fscheiner** [comment]
@enricovianello 
Sorry for the delay, I had already created a comment but it seems it got lost somehow.

> @fscheiner No, we aren't building our GridFTP server from sources. We developed a dsi module. So, If you've already built a rpm of globus-gridftp-server (something like a nightly or a beta rpm) we can easily test it.

Sorry, we don't do that right now, but only create updated source tarballs with each new release. But thanks to @matyasselmeci's scripts you can create RPMs yourself from within the source tree. Here's a quick guide on how to do that (I used CentOS7 but assume this will also work for CentOS6 and might also work for CentOS8).

1. First install packages needed for the build:
```
$ sudo yum install make autoconf automake libtool libtool-ltdl-devel patch curl git bison openssl openssl-devel rpm-build doxygen graphviz 'perl(Pod::Html)' fakeroot udt udt-devel glib2-devel libnice-devel gettext-devel libffi-devel libxml2-devel pam-devel voms-devel cyrus-sasl-devel openldap-devel voms-clients initscripts python-devel 'perl(DBI)' redhat-lsb-core m2crypto mod_ssl mod_wsgi pyOpenSSL python-crypto perl-generators 'perl(Test::More)' gcc-c++ 'perl(URI)'
```

2. Then enter the `<SOURCE_TREE_ROOT>/travis-ci/` subdirectory of the source tree and invoke the following commands:
```
$ ./make_source_tarballs.sh
$ ./make_rpms.sh -C
```
> **NOTICE:** If the make process fails due to missing dependencies, install the dependencies and restart the failed command. I did my tests in a non-pristine CentOS7 driven VM, so don't know for sure, that the above package list is complete.

3. The resulting RPM packages are available in `<SOURCE_TREE_ROOT>/packaging/rpmbuild/RPMS/`

**Comment by fscheiner** [comment]
Closed with [GCT 6.2.20201212 maintenance release](https://github.com/gridcf/gct/releases/tag/v6.2.20201212).
